src/resources/troubleshooting.md

```diff
@@ -0,0 +1,771 @@
+
+**The 2nd Reason**\
+Your bot was rate limited. This can happen because of excessive requests to Discord API, performed by BDFD functions (such as `$addEmoji` and others).\
+You should not stack such functions and try to perform them all at once.
+
+**The 3rd Reason**\
+The node your bot is running on is experiencing some slight problems (such as a rate limits). If you're sure this is the case, you can regenerate the bot's token and replace it with a new one. This will cause your bot to change its node.
+
+### The Slash Commands Doesn't Appear
+**The 1st Reason**\
+Old version of the application. Make sure that you have a latest version of the application installed.\
+New versions of the application have been improved and updated, and new features have been added. In addition, errors with the validity check of the slash commands have been fixed. You will be warned if you set up your slash command incorrectly, in which case the application will not allow you to save the slash command.
+
+**The 2nd Reason**\
+Slash commands are cached by discord, so it takes time before they appear in discord.
+
+They're also cached on the client side, if they were successfully cached in discord. If other users have a new slash command and you don't, restart the discord client.\
+Restarting will cause existing slash commands to be cleared and new ones will cache.
+
+**The 3rd Reason**\
+Conflict of slash commands due to other services that you're no longer using. For example, if two slash commands of same type have the same name, but one is created using a third-party service, and the other through the BDFD, this can cause a conflict and the slash command will not appear.\
+You can solve this problem by Syncing slash commands with discord in the bot's settings. This removes third-party service slash commands and leaves only those that were created in our application.
+    
+![sync](https://user-images.githubusercontent.com/70456337/200124426-6dbe69aa-1711-4838-9569-689b5e30f48f.gif)
+
+### Integration Requires Code Grant
+You can only get this error when trying to invite a bot to any server.\
+Most likely, you have accidentally or unknowingly enabled the `Require OAauth2 Code Grant` parameter in your bot's settings in the [developer portal](https://discord.com/developers/applications).\
+This is the reason why you get this error.
+
+This parameter is required only for applications with scopes such as identify, email, and others to work with the user account in discord. But BDFD doesn't have such support, so you should not enable this parameter or choose any other scopes other than `bot`.
+
+---
+## App Issues
+
+### The Ad Button Doesn't Work
+> In this section, you can skip possible reasons to [Possible Solutions](#possible-solutions).
+
+**The 1st Reason**\
+Unstable Internet connection. Make sure your Internet connection is stable and not too slow, because you have to load the ad first to watch it. This is why you see " Loading ad...".
```

Comment:
```suggestion
Unstable Internet connection. Make sure your Internet connection is stable and not too slow, because you have to load the ad first to watch it. This is why you see "Loading ad...".
```